src/language-js/printer-estree.js

```diff
@@ -5750,7 +5750,10 @@ function printBinaryishExpressions(
 
 function printAssignmentRight(leftNode, rightNode, printedRight, options) {
   if (hasLeadingOwnLineComment(options.originalText, rightNode, options)) {
-    return indent(concat([hardline, printedRight]));
+    const shouldPrintLine = rightNode.comments.some(
+      handleComments.isBlockComment
+    );
+    return indent(concat([shouldPrintLine ? line : hardline, printedRight]));
```

Comment:
Can't we always use `line` here? If there is a line comment, it'll add its own `hardline` anyway, so the group will break and `line` will be printed as a new line.